react/the_react_ecosystem/fetching-data.md

```diff
@@ -0,0 +1,297 @@
+### Introduction
+
+Up to this point we have been using React to build client side applications with interactive user interfaces, but what if we want fetch data from the internet? In order to create full fledged web applications, we need some way to get data from external sources and dynamically displaying it on the client.
+
+In this lesson, we'll explore the ins and outs of fetching data in React, starting with the basics of making API calls, managing component state, and handling asynchronous operations using JavaScript's ~fetch~ function.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Understand how to make fetch requests in React components.
+- Using conditonals to render content depending on the state of the request (short circuiting).
+- Catching and handling errors.
+- Lifting requests up the component hierarchy.
+
+### A basic fetch request
+
+Before we dive into the specifics of fetching data in React, let's first revisit how we can use the fetch API to get data from a server.
+
+<!-- fetch is used to make requests to external APIs and returns a Promise that resolves to the response object representing the server's response. -->
```

Comment:
I don't think we need this comment. Learners are already familiar with fetch and promises
```suggestion
```